model/docs/extending_pyrenew.qmd

```diff
@@ -0,0 +1,256 @@
+        inf_feedback_pmf, *_ = self.infection_feedback_pmf.sample(**kwargs)
+        inf_feedback_pmf = du.pad_x_to_match_y(x=inf_feedback_pmf, y=Rt)
+
+        # Generating the infections with feedback
+        all_infections, Rt_adj = compute_infections_from_rt_with_feedback(
+            I0=I0_vec,
+            Rt_raw=Rt,
+            infection_feedback_strength=inf_feedback_strength,
+            reversed_generation_interval_pmf=gen_int_rev,
+            reversed_infection_feedback_pmf=inf_feedback_pmf,
+        )
+
+        # Storing adjusted Rt for future use
+        npro.deterministic("Rt_adjusted", Rt_adj)
+
+        # Preparing theoutput
+
+        return InfFeedbackSample(
+            infections=all_infections,
+            rt=Rt_adj,
+            )
+```
+
+The core of the class is implemented in the `sample()` method. Things to highlight from the above code:
+
+1. **Arguments of `sample`**: The `InfFeedback` class will be used within `RtInfectionsRenewalModel` to generate latent infections. During the sampling process, `InfFeedback.sample()` will receive the reproduction number, the initial number of infections, and the generation interval. `RandomVariable.sample()` calls are expected to include the `**kwargs` argument, even if unused.
+
+2. **Calls to `RandomVariable.sample()`**: All calls to `RandomVariable.sample()` are expected to return a tuple or named tuple. In our implementation, we capture the output of `infection_feedback_strength.sample()` and `infection_feedback_pmf.sample()` in the variables `inf_feedback_strength` and `inf_feedback_pmf`, respectively, disregarding the other outputs (i.e., using `*_`).
+
+3. **Saving computed quantities**: Since `Rt_adj` is not generated via `numpyro.sample()`, we use `numpyro.deterministic()` to record the quantity to a site; allowing us to access it later.
+
+4. **Return type of `InfFeedback.sample()`**: As said before, the `sample()` method should return a tuple or named tuple. In our case, we return a named tuple `InfFeedbackSample` with two fields: `infections` and `rt`.
+
+```{python}
+#| label: simulation2
+latent_infections2 = InfFeedback(
+  infection_feedback_strength = gen_int,
+  infection_feedback_pmf = gen_int,
+)
+
+model1 = RtInfectionsRenewalModel(
+    gen_int=gen_int,
+    I0=I0,
+    latent_infections=latent_infections2,
+    Rt_process=rt,
+    observation_process=None,
+)
+
+# Sampling and fitting model 0 (with no obs for infections)
+np.random.seed(223)
+with npro.handlers.seed(rng_seed=np.random.randint(1, 600)):
+    model1_samp = model1.sample(n_timepoints=30)
+```
+
+Comparing `model0` with `model1`:
+
+```{python}
+#| label: fig-simulate2
+#| fig-cap: Simulated infections with no observation process
+import matplotlib.pyplot as plt
+fig, ax = plt.subplots(ncols=2)
+ax[0].plot(model0_samp.latent_infections)
+ax[1].plot(model1_samp.latent_infections)
+ax[0].set_xlabel("Time")
+ax[1].set_xlabel("Time")
+ax[0].set_ylabel("Infections")
+plt.show()
+```
```

Comment:
label the figures with `model0` and `model1`

Reply:
Addressed in 5eb31cc